packages/astro/test/fixtures/integration-call-hook/astro.config.ts

```diff
@@ -0,0 +1,46 @@
+import { defineConfig } from 'astro/config'
+import {} from 'astro'
+
+type Config = {
+	foo: string;
+}
+
+interface Hooks {
+	'a:config'?: (options: { config: Config }) => void
+}
+
+declare module 'astro' {
+	interface AstroIntegrationsHooks extends Hooks {}
+}
+
+export default defineConfig({
+	integrations: [
+		{
+			name: 'a',
+			hooks: {
+				'astro:config:setup': ({ callHook }) => {
+					callHook('a:config', { config: { foo: 'bar' }})
```

Comment:
Module augmentation does not seem to work here, ideas welcome

Reply:
`callHook` is async so you might need to use `await` here (I didn't do a thorough review yet, so it might not be all it takes)